grails-app/services/au/org/ala/ecodata/ActivityService.groovy

```diff
@@ -367,4 +393,16 @@ class ActivityService {
         activities.collect{toMap(it, levelOfDetail)}
     }
 
+    def getAllActivityIdsForProjectActivity(String pActivityId) {
```

Comment:
This can be rewritten as 
```
List<String> getAll...(String projectActivityId) {
  Activity.withCriteria {
    eq "projectActivityId", projectActivityId
    ne "status", DELETED

    projections {
        property("activityId")
    }
}
```

The *.toString is not necessary: the projection will return a list of strings.